x/datadeal/abci.go

```diff
@@ -1,13 +1,66 @@
 package datadeal
 
 import (
+	"strconv"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/medibloc/panacea-core/v2/x/datadeal/keeper"
 	"github.com/medibloc/panacea-core/v2/x/datadeal/types"
 	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
 )
 
 func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
+	keeper.IterateClosedDataVerificationQueue(ctx, ctx.BlockHeader().Time, func(dataSale *types.DataSale) bool {
+
+		keeper.RemoveDataVerificationQueue(ctx, dataSale.DealId, dataSale.VerifiableCid, dataSale.VotingPeriod.VotingEndTime)
+		iterator := keeper.GetDataVerificationVoteIterator(ctx, dataSale.DealId, dataSale.VerifiableCid)
+
+		defer iterator.Close()
+
+		oracleKeeper := keeper.GetOracleKeeper()
+
+		tallyResult, err := oracleKeeper.Tally(
+			ctx,
+			iterator,
+			func() oracletypes.Vote {
+				return &types.DataVerificationVote{}
+			},
+			func(vote oracletypes.Vote) error {
+				return keeper.RemoveDataVerificationVote(ctx, vote.(*types.DataVerificationVote))
+			},
+		)
+
+		if err != nil {
+			panic(err)
+		}
+
+		if tallyResult.IsPassed() {
+			dataSale.Status = types.DATA_SALE_STATUS_DELIVERY_VOTING_PERIOD
+			dataSale.VerifiableCid = string(tallyResult.ConsensusValue)
```

Comment:
I think it would be better to compare the verifiableCID of the consensus verifiableCID with the verifiableCID of dataSale if the vote passed.
Because verifiableCID is already stored in dataSale.

Reply:
Actually, there is no consensus value in data verification vote, right?

Reply:
When the issue #448 resolved, the data verification vote will not have consensus value.

Reply:
After this branch is merged, I will reflect this part in the PR that adds dataHash (#448).

Reply:
Applied in [b57333c](https://github.com/medibloc/panacea-core/pull/449/commits/b57333c3f6321de4fa9afa74f420923362cc5b3a).